### ISSUE TITLE
Add a `no_empty_line_after_inner_attr` lint

### ISSUE THREAD

**Issue by xtqqczze** [rust-lang/rust-clippy#17687]
### What it does

Warns when an inner attribute (`#![attr]`) is immediately followed by a non-empty line without a blank line in between.

This lint would be part of the `suspicious` category.

### Advantage

* Improves readability by visually separating attributes from the code they apply to. Without the blank line, an inner attribute can look like it was intended to be an outer attribute (`#[attr]`). If it is intended to be an inner attribute, a blank line should separate it from the following code.

### Drawbacks

* Primarily a formatting/style issue with no effect on correctness.

### Example

```rust
#![allow(unused_variables)]
fn foo() {
    let x = 1;
}

fn bar() {
    let x = 1;
}
```

Could be written as:

```rust
#![allow(unused_variables)]

fn foo() {
    let x = 1;
}

fn bar() {
    let x = 1;
}
```

### Comparison with existing lints

The closest existing lint is ⁠[`empty_line_after_outer_attr`](https://rust-lang.github.io/rust-clippy/master/#empty_line_after_outer_attr), which checks for an empty line after an outer attribute. This lint would provide the corresponding behavior for inner attributes.

### Additional Context

* rustfmt’s unstable [`blank_lines_lower_bound`](https://rust-lang.github.io/rustfmt/?search=&version=v1.10.0#blank_lines_lower_bound) does not provide a way to enforce a blank line specifically after an inner attribute.

**Comment by CommanderStorm** [comment]
Would this not be something we want in rusftfmt instead? 
I cannot quite imagine that someone might actually prefer module attr where an fn attr would go. That is just confusing...

**Comment by xtqqczze** [comment]
@CommanderStorm I've updated the description for clarification.

**Comment by CommanderStorm** [comment]
<img width="968" height="471" alt="Image" src="https://github.com/user-attachments/assets/173b7d04-f872-42a7-8154-aca7c4665484" />

> rustfmt’s unstable [blank_lines_lower_bound](https://rust-lang.github.io/rustfmt/?search=&version=v1.10.0#blank_lines_lower_bound).

Can't say it works well currently.

In any case, I think this is not something that SHOULD be clipppy, but rustfmt unless there is a very good reason for it, since this feels to me pretty much like formatting, not linting..

I think the effort being spend on improving rustfmt would be much more valuable.

**Comment by xtqqczze** [comment]
Well, we already have [`clippy::empty_line_after_outer_attr`](https://rust-lang.github.io/rust-clippy/master/#empty_line_after_outer_attr) in the `suspicious` category, so it would be nice to have the corresponding lint for inner attributes as well.

I also don’t think this is _purely_ a formatting issue: without the blank line, an inner attribute can easily be mistaken for an outer attribute, which hurts readability. Since this lint is specifically about preventing that ambiguity, I think this also is a good fit in Clippy’s `suspicious` category.
